test/ng/compileSpec.js

```diff
@@ -5629,6 +5629,34 @@ describe('$compile', function() {
       });
 
 
+      //see issue https://github.com/angular/angular.js/issues/12936
+      it('should use the proper scope when being the root element of a replaced directive', function() {
```

Comment:
I afree we probably don't need a test for `templateUrl` but what we do need a test for is child scope (i.e. not isolated scope), since this is a common failing case (e.g. `ngIf` or `ngRepeat` at the root of the template).